src/richie/apps/search/indexers/subjects.py

```diff
@@ -55,3 +57,40 @@ def format_es_subject_for_api(es_subject, best_language):
                 es_subject["_source"]["name"], best_language
             ),
         }
+
+    @staticmethod
+    def build_es_query(request):
+        """
+        Build an ElasticSearch query and its related aggregations, to be consumed by the ES client
+        in the Subjects ViewSet
+        """
+        # Instantiate a form with our query_params to check & sanitize them
+        params_form = SubjectListForm(request.query_params)
+
+        # Raise an exception with error information if the query params are not valid
+        if not params_form.is_valid():
+            raise QueryFormatException(params_form.errors)
+
+        # Build a query that matches on the name field if it was handed by the client
+        # Note: test_elasticsearch_feature.py needs to be updated whenever the search call
+        # is updated and makes use new features.
+        if params_form.cleaned_data.get("query"):
+            query = {
+                "query": {
+                    "match": {
+                        "name.fr": {
+                            "query": params_form.cleaned_data.get("query"),
+                            "analyzer": "french",
```

Comment:
it looks like this should be configurable but maybe not in the frame of this PR.